frontend/model/contracts/group.js

```diff
@@ -908,6 +908,11 @@ sbp('chelonia/defineContract', {
         const { username } = rootState.loggedIn
 
         if (data.member === username) {
+          // NOTE: should remove archived data from IndexedStorage
+          //       regarding the current group (proposals, payments)
+          sbp('gi.contracts/group/removeArchivedProposals', contractID)
+          sbp('gi.contracts/group/removeArchivedPayments', contractID)
+
```

Comment:
1. I think this should be below, after the check for `'JOINING_GROUP-'`
2. These are async methods that should be await'd on, and therefore the `sideEffect` should also be rewritten to be `async sideEffect`

Reply:
Nice catch. :+1: